ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -192,31 +205,47 @@ def get_pcf_channel_metrics(self, queue_manager, tags, config):
             self.gauge(mname, channels, tags=tags)
 
         # grab all the discoverable channels
-        self._get_channel_status(queue_manager, '*', tags, config)
+        self._submit_channel_status(queue_manager, '*', tags, config)
 
         # check specific channels as well
         # if a channel is not listed in the above one, a user may want to check it specifically,
         # in this case it'll fail
         for channel in config.channels:
-            self._get_channel_status(queue_manager, channel, tags, config)
+            self._submit_channel_status(queue_manager, channel, tags, config)
 
-    def _get_channel_status(self, queue_manager, channel, tags, config):
-        channel_tags = tags + ["channel:{}".format(channel)]
+    def _submit_channel_status(self, queue_manager, search_channel_name, tags, config):
+        """Submit channel status
+        :param search_channel_name might contain wildcard characters
+        """
         try:
-            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(channel)}
+            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(search_channel_name)}
             pcf = pymqi.PCFExecute(queue_manager)
             response = pcf.MQCMD_INQUIRE_CHANNEL_STATUS(args)
         except pymqi.MQMIError as e:
             self.log.warning("Error getting CHANNEL stats {}".format(e))
+            channel_tags = tags + ["channel:{}".format(search_channel_name)]
             self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.CRITICAL, channel_tags)
         else:
             for channel_info in response:
-                name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME]
-                name = name.strip()
+                channel_name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME].decode().strip()
+                channel_tags = tags + ["channel:{}".format(channel_name)]
 
-                # running = 3, stopped = 4
-                status = channel_info[pymqi.CMQCFC.MQIACH_CHANNEL_STATUS]
-                if status == 3:
+                channel_status = channel_info[pymqi.CMQCFC.MQIACH_CHANNEL_STATUS]
+
+                if channel_status == pymqi.CMQCFC.MQCHS_RUNNING:
                     self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.OK, channel_tags)
-                elif status == 4:
+                elif channel_status == pymqi.CMQCFC.MQCHS_STOPPING:
```

Comment:
I'm surprised we're not submitting anything for the rest of statuses but not the focus of this PR

Reply:
The PR is still in draft, you are reviewing a bit early :)

Yes, I'm also a bit surprised by this, not sure why, usually I tried by map every status to a Datadog service check available statuses.

Reply:
I will remap this to cover all statuses.

Reply:
@dsahni @hithwen 

1) There is a small change on the behavior of the service check `ibm_mq.channel` (can connect to channel).
**Before**, `ibm_mq.channel` status check was verifying that the channel can connect, but also partially using the status (OK if RUNNING, WARNING if STOPPING). And if the status it something else than RUNNING and STOPPING, nothing was reported.
**Now**, `ibm_mq.channel` will report OK if it can connect and CRITICAL otherwise, as advertised by it's current (no change):
```json
    {
        "agent_version": "6.8.1",
        "integration":"IBM MQ",
        "groups": ["queue_manager", "mq_host", "port", "channel"],
        "check": "ibm_mq.channel",
        "statuses": ["ok", "critical"],
        "name": "Channel Can Connect",
        "description": "Returns `CRITICAL` if the Agent cannot retrieve channel stats. Returns `OK` otherwise."
    }
```

2) A new service have been created to monitor status.
```json
    {
        "agent_version": "6.8.1",
        "integration":"IBM MQ",
        "groups": ["queue_manager", "mq_host", "port", "channel"],
        "check": "ibm_mq.channel.status",
        "statuses": ["ok", "critical", "warning", "unknown"],
        "name": "Channel Status",
        "description": "Return `CRITICAL` if the status is INACTIVE/STOPPED/STOPPING. Returns `OK` if the status is RUNNING. Returns `WARNING` if the status might lead to running."
    }
```

Some redundancy, since we can achieve similar things with the newly added metric:
```
ibm_mq.channel.count,gauge,,,,Sum by status to count channels. Filter by channel and status tags to create notifications.,0,ibm_mq,channel count
```

But I guess, it's OK to have some redundancy like this one, the user can pick the one that suits the most his needs.